websocket_test.go

```diff
@@ -22,25 +23,48 @@ func TestCanDial(t *testing.T) {
 	if err != nil {
 		t.Fatal(err)
 	}
+	addrWss, err := ma.NewMultiaddr("/dns4/example.com/tcp/5555/wss")
+	if err != nil {
+		t.Fatal(err)
+	}
 
 	addrTCP, err := ma.NewMultiaddr("/ip4/127.0.0.1/tcp/5555")
 	if err != nil {
 		t.Fatal(err)
 	}
 
 	d := &WebsocketTransport{}
-	matchTrue := d.CanDial(addrWs)
-	matchFalse := d.CanDial(addrTCP)
-
-	if !matchTrue {
+	if !d.CanDial(addrWs) {
 		t.Fatal("expected to match websocket maddr, but did not")
 	}
-
-	if matchFalse {
+	if !d.CanDial(addrWss) {
+		t.Fatal("expected to match secure websocket maddr, but did not")
+	}
+	if d.CanDial(addrTCP) {
 		t.Fatal("expected to not match tcp maddr, but did")
 	}
 }
 
+func TestDialWss(t *testing.T) {
+	raddr, err := ma.NewMultiaddr("/dns4/nyc-1.bootstrap.libp2p.io/tcp/443/wss")
```

Comment:
Is it okay to leave this in the tests? Usually I would say it is not proper to dial out to an independent server in this kind of test, but it did help me find some bugs. Without this, we cannot easily test that dialing actually works.

Reply:
It'll make this test fail in offline scenarios. I rather test whether the host is resolvable, and if not, `t.Skip()` the test. Do we have a ping-pong test? (libp2p WSS listener and dialer interacting with one another)

Reply:
Since I didn't implement listening in this PR, it is pretty hard to write a ping-pong test the way we normally would (e.g. with two hosts running locally). Even if we did have listening there would be some challenges because we would have to use a domain name with an SSL certificate.

My vote is to `t.Skip()` this test for now and manually enable it whenever we need to test dialing again in the future. What do you think?

Reply:
@raulk I updated the tests so that if we can't resolve the host, we call `t.Skip`. I also improved the test by opening a stream to make sure that the connection is _really_ working.